File: api/elo.py
import math

def apply_result(current_elo, coolness_score):
    # Simple ELO implementation for summer adventures
    # Base ELO: 1200
    # Gain: +10 per 10 coolness points (max +100 at coolness 100)
    elo_gain = math.ceil((coolness_score / 10) * 10)
    return max(800, current_elo + elo_gain)

def rank_players(players):
    return sorted(players, key=lambda p: p.get('elo', 1200), reverse=True)

File: lib/elo.js
// Simple ELO-style rating for summer adventures.
//
// A coolness of 50 is "par" (no change). Above 50 you gain, below you lose,
// scaled by a K-factor. Epic trip (100) => +K, lame proof (0) => -K.
// Rating is floored at 800 so nobody bottoms out.

const BASELINE_COOLNESS = 50;
const K_FACTOR = 40;
const FLOOR = 800;

export function applyResult(currentElo, coolnessScore) {
  const clamped = Math.min(100, Math.max(0, Number(coolnessScore) || 0));
  const delta = Math.round((K_FACTOR * (clamped - BASELINE_COOLNESS)) / BASELINE_COOLNESS);
  return Math.max(FLOOR, Math.round(currentElo) + delta);
}

export function rankPlayers(players) {
  return [...players].sort((a, b) => b.elo - a.elo);
}
